{
  "title": "Android 14’s storage disaster gets patched, but your data might be gone",
  "publishedAt": "2023-11-06T22:00:00.000Z",
  "updatedAt": "2023-11-06T22:00:00.000Z",
  "description": "We get sold technical explainers for all these features, but when they were really needed, none of these poorly thought-out, half-baked systems worked",
  "image": {
    "filePath": "../public/blogs/android-rage-out.jpeg",
    "relativeFilePath": "../../public/blogs/android-rage-out.jpeg",
    "format": "jpeg",
    "height": 360,
    "width": 640,
    "aspectRatio": 1.7777777777777777,
    "blurhashDataUrl": "data:image/jpeg;base64,/9j/2wBDAAoHBwgHBgoICAgLCgoLDhgQDg0NDh0VFhEYIx8lJCIfIiEmKzcvJik0KSEiMEExNDk7Pj4+JS5ESUM8SDc9Pjv/2wBDAQoLCw4NDhwQEBw7KCIoOzs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozv/wAARCAAIAAgDASIAAhEBAxEB/8QAFQABAQAAAAAAAAAAAAAAAAAAAAT/xAAdEAABBAIDAAAAAAAAAAAAAAABAAIDEQQFBiFx/8QAFQEBAQAAAAAAAAAAAAAAAAAABQb/xAAfEQEAAAQHAAAAAAAAAAAAAAABAAIDEQQFITFRYZH/2gAMAwEAAhEDEQA/AIjyHbDMyI3xGWYFxEZprTYu7PdeoiI2pjqstttQYp3L6CCFvOXqP//Z"
  },
  "isPublished": true,
  "author": "Mark Sikaundi",
  "tags": [
    "apple"
  ],
  "body": {
    "raw": "\nIt's the start of November, and that means a new Android security patch. Google claims this one is fixing a high-profile Android 14 storage bug that was locking some people out of their devices. The November Security Bulletin contains the usual pile of security fixes, while consumer-oriented Pixel patch notes list a few user-facing changes. The important line is \"Fix for issue occasionally causing devices with multiple users enabled to show out of space or be in a reboot loop.\" A footnote points out that this is for the \"Pixel 6, Pixel 6a, 6 Pro, 7, 7 Pro, 7a, Tablet, Fold, Pixel 8, Pixel 8 Pro.\"\n\nWe're on about day 33 of the Android 14 storage bug. For devices with multiple users set up, there is some kind of storage issue that is locking users out of their device. Some are completely unusable, with the phone bootlooping constantly and never reaching the home screen. Others are able to boot up the device but don't have access to lock storage, which causes a huge amount of issues. Some users likened the bug to \"ransomware,\" a type of malware that encrypts your local storage and then demands money for your data. One fix is to completely erase your device with a factory reset, but a lot of users don't want to do that.\n\nAndroid 14’s user-profile data bug seems indistinguishable from ransomware\n\nThe earliest reports of this started just days after the October 4 launch date. Google usually rolls updates out slowly so it can pull them if issues like this pop up to minimize damage. That didn't happen here, though. Google failed to respond quickly to initial reports and just let the bug roll out to everyone. Some people even report being freshly hit with the bug just four days ago because Google 1) let the update roll out without stopping it and 2) can't patch its software quickly enough. The biggest issue tracker thread on this bug is up to 1,000-plus likes and 850 comments of people locked out of their devices, and it took two separate rounds of news coverage for Google to acknowledge the bug after about 20 days.\n\nGoogle promised a rescue patch, which is now arriving, though the company said only \"some\" data would be recoverable and that \"this update may not enable data to be recovered for devices that are repeatedly rebooting.\" That seems to line up with what people are experiencing now that the OTA is out. People who can boot up report fixed devices, while bootloopers aren't getting anywhere. Some users are still in the support thread hoping for some kind of data recovery for bootlooping devices, but it doesn't seem like that is going to happen.\n\nThis whole fiasco has been a complete failure of most of the controls and protections Google has in place in Android. The company slowly rolls out updates to stop problems before it hits a wide number of users, but it failed to pull the update when problems arose. Android has dual system partitions so that you always have a backup if the device fails to boot after an update, but that system didn't work here because Google's \"boot failure\" detection isn't accurate enough. The company shipped a quick-fix patch via Google Play System Updates in the Play Store, but because those passively wait around for a reboot to get applied, users still got hit by the bug days after that patch came out. Android is supposed to have a data backup system for apps, but because that doesn't work well and isn't forced on every app, many users have no backups at all.\n\nWe get sold technical explainers for all these features, but when they were really needed, none of these poorly thought-out, half-baked systems worked. This disaster is a complete technical failure of several Android systems, and many changes need to happen.\n",
    "code": "var Component=(()=>{var ur=Object.create;var N=Object.defineProperty;var lr=Object.getOwnPropertyDescriptor;var cr=Object.getOwnPropertyNames;var fr=Object.getPrototypeOf,dr=Object.prototype.hasOwnProperty;var q=(l,i)=>()=>(i||l((i={exports:{}}).exports,i),i.exports),pr=(l,i)=>{for(var v in i)N(l,v,{get:i[v],enumerable:!0})},Ee=(l,i,v,_)=>{if(i&&typeof i==\"object\"||typeof i==\"function\")for(let y of cr(i))!dr.call(l,y)&&y!==v&&N(l,y,{get:()=>i[y],enumerable:!(_=lr(i,y))||_.enumerable});return l};var hr=(l,i,v)=>(v=l!=null?ur(fr(l)):{},Ee(i||!l||!l.__esModule?N(v,\"default\",{value:l,enumerable:!0}):v,l)),mr=l=>Ee(N({},\"__esModule\",{value:!0}),l);var ke=q((Er,we)=>{we.exports=React});var xe=q(z=>{\"use strict\";(function(){\"use strict\";var l=ke(),i=Symbol.for(\"react.element\"),v=Symbol.for(\"react.portal\"),_=Symbol.for(\"react.fragment\"),y=Symbol.for(\"react.strict_mode\"),X=Symbol.for(\"react.profiler\"),K=Symbol.for(\"react.provider\"),H=Symbol.for(\"react.context\"),P=Symbol.for(\"react.forward_ref\"),F=Symbol.for(\"react.suspense\"),I=Symbol.for(\"react.suspense_list\"),O=Symbol.for(\"react.memo\"),U=Symbol.for(\"react.lazy\"),Oe=Symbol.for(\"react.offscreen\"),J=Symbol.iterator,Se=\"@@iterator\";function Ce(e){if(e===null||typeof e!=\"object\")return null;var r=J&&e[J]||e[Se];return typeof r==\"function\"?r:null}var w=l.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;function d(e){{for(var r=arguments.length,t=new Array(r>1?r-1:0),n=1;n<r;n++)t[n-1]=arguments[n];je(\"error\",e,t)}}function je(e,r,t){{var n=w.ReactDebugCurrentFrame,s=n.getStackAddendum();s!==\"\"&&(r+=\"%s\",t=t.concat([s]));var u=t.map(function(o){return String(o)});u.unshift(\"Warning: \"+r),Function.prototype.apply.call(console[e],console,u)}}var De=!1,Ae=!1,Ne=!1,Fe=!1,Ie=!1,Z;Z=Symbol.for(\"react.module.reference\");function Ue(e){return!!(typeof e==\"string\"||typeof e==\"function\"||e===_||e===X||Ie||e===y||e===F||e===I||Fe||e===Oe||De||Ae||Ne||typeof e==\"object\"&&e!==null&&(e.$$typeof===U||e.$$typeof===O||e.$$typeof===K||e.$$typeof===H||e.$$typeof===P||e.$$typeof===Z||e.getModuleId!==void 0))}function We(e,r,t){var n=e.displayName;if(n)return n;var s=r.displayName||r.name||\"\";return s!==\"\"?t+\"(\"+s+\")\":t}function Q(e){return e.displayName||\"Context\"}function b(e){if(e==null)return null;if(typeof e.tag==\"number\"&&d(\"Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue.\"),typeof e==\"function\")return e.displayName||e.name||null;if(typeof e==\"string\")return e;switch(e){case _:return\"Fragment\";case v:return\"Portal\";case X:return\"Profiler\";case y:return\"StrictMode\";case F:return\"Suspense\";case I:return\"SuspenseList\"}if(typeof e==\"object\")switch(e.$$typeof){case H:var r=e;return Q(r)+\".Consumer\";case K:var t=e;return Q(t._context)+\".Provider\";case P:return We(e,e.render,\"ForwardRef\");case O:var n=e.displayName||null;return n!==null?n:b(e.type)||\"Memo\";case U:{var s=e,u=s._payload,o=s._init;try{return b(o(u))}catch{return null}}}return null}var E=Object.assign,R=0,ee,re,te,ne,ae,oe,ie;function se(){}se.__reactDisabledLog=!0;function Ye(){{if(R===0){ee=console.log,re=console.info,te=console.warn,ne=console.error,ae=console.group,oe=console.groupCollapsed,ie=console.groupEnd;var e={configurable:!0,enumerable:!0,value:se,writable:!0};Object.defineProperties(console,{info:e,log:e,warn:e,error:e,group:e,groupCollapsed:e,groupEnd:e})}R++}}function $e(){{if(R--,R===0){var e={configurable:!0,enumerable:!0,writable:!0};Object.defineProperties(console,{log:E({},e,{value:ee}),info:E({},e,{value:re}),warn:E({},e,{value:te}),error:E({},e,{value:ne}),group:E({},e,{value:ae}),groupCollapsed:E({},e,{value:oe}),groupEnd:E({},e,{value:ie})})}R<0&&d(\"disabledDepth fell below zero. This is a bug in React. Please file an issue.\")}}var W=w.ReactCurrentDispatcher,Y;function S(e,r,t){{if(Y===void 0)try{throw Error()}catch(s){var n=s.stack.trim().match(/\\n( *(at )?)/);Y=n&&n[1]||\"\"}return`\n`+Y+e}}var $=!1,C;{var Me=typeof WeakMap==\"function\"?WeakMap:Map;C=new Me}function ue(e,r){if(!e||$)return\"\";{var t=C.get(e);if(t!==void 0)return t}var n;$=!0;var s=Error.prepareStackTrace;Error.prepareStackTrace=void 0;var u;u=W.current,W.current=null,Ye();try{if(r){var o=function(){throw Error()};if(Object.defineProperty(o.prototype,\"props\",{set:function(){throw Error()}}),typeof Reflect==\"object\"&&Reflect.construct){try{Reflect.construct(o,[])}catch(g){n=g}Reflect.construct(e,[],o)}else{try{o.call()}catch(g){n=g}e.call(o.prototype)}}else{try{throw Error()}catch(g){n=g}e()}}catch(g){if(g&&n&&typeof g.stack==\"string\"){for(var a=g.stack.split(`\n`),p=n.stack.split(`\n`),c=a.length-1,f=p.length-1;c>=1&&f>=0&&a[c]!==p[f];)f--;for(;c>=1&&f>=0;c--,f--)if(a[c]!==p[f]){if(c!==1||f!==1)do if(c--,f--,f<0||a[c]!==p[f]){var m=`\n`+a[c].replace(\" at new \",\" at \");return e.displayName&&m.includes(\"<anonymous>\")&&(m=m.replace(\"<anonymous>\",e.displayName)),typeof e==\"function\"&&C.set(e,m),m}while(c>=1&&f>=0);break}}}finally{$=!1,W.current=u,$e(),Error.prepareStackTrace=s}var x=e?e.displayName||e.name:\"\",_e=x?S(x):\"\";return typeof e==\"function\"&&C.set(e,_e),_e}function Ve(e,r,t){return ue(e,!1)}function Le(e){var r=e.prototype;return!!(r&&r.isReactComponent)}function j(e,r,t){if(e==null)return\"\";if(typeof e==\"function\")return ue(e,Le(e));if(typeof e==\"string\")return S(e);switch(e){case F:return S(\"Suspense\");case I:return S(\"SuspenseList\")}if(typeof e==\"object\")switch(e.$$typeof){case P:return Ve(e.render);case O:return j(e.type,r,t);case U:{var n=e,s=n._payload,u=n._init;try{return j(u(s),r,t)}catch{}}}return\"\"}var D=Object.prototype.hasOwnProperty,le={},ce=w.ReactDebugCurrentFrame;function A(e){if(e){var r=e._owner,t=j(e.type,e._source,r?r.type:null);ce.setExtraStackFrame(t)}else ce.setExtraStackFrame(null)}function Ge(e,r,t,n,s){{var u=Function.call.bind(D);for(var o in e)if(u(e,o)){var a=void 0;try{if(typeof e[o]!=\"function\"){var p=Error((n||\"React class\")+\": \"+t+\" type `\"+o+\"` is invalid; it must be a function, usually from the `prop-types` package, but received `\"+typeof e[o]+\"`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.\");throw p.name=\"Invariant Violation\",p}a=e[o](r,o,n,t,null,\"SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED\")}catch(c){a=c}a&&!(a instanceof Error)&&(A(s),d(\"%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).\",n||\"React class\",t,o,typeof a),A(null)),a instanceof Error&&!(a.message in le)&&(le[a.message]=!0,A(s),d(\"Failed %s type: %s\",t,a.message),A(null))}}}var Be=Array.isArray;function M(e){return Be(e)}function qe(e){{var r=typeof Symbol==\"function\"&&Symbol.toStringTag,t=r&&e[Symbol.toStringTag]||e.constructor.name||\"Object\";return t}}function ze(e){try{return fe(e),!1}catch{return!0}}function fe(e){return\"\"+e}function de(e){if(ze(e))return d(\"The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.\",qe(e)),fe(e)}var T=w.ReactCurrentOwner,Xe={key:!0,ref:!0,__self:!0,__source:!0},pe,he,V;V={};function Ke(e){if(D.call(e,\"ref\")){var r=Object.getOwnPropertyDescriptor(e,\"ref\").get;if(r&&r.isReactWarning)return!1}return e.ref!==void 0}function He(e){if(D.call(e,\"key\")){var r=Object.getOwnPropertyDescriptor(e,\"key\").get;if(r&&r.isReactWarning)return!1}return e.key!==void 0}function Je(e,r){if(typeof e.ref==\"string\"&&T.current&&r&&T.current.stateNode!==r){var t=b(T.current.type);V[t]||(d('Component \"%s\" contains the string ref \"%s\". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref',b(T.current.type),e.ref),V[t]=!0)}}function Ze(e,r){{var t=function(){pe||(pe=!0,d(\"%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)\",r))};t.isReactWarning=!0,Object.defineProperty(e,\"key\",{get:t,configurable:!0})}}function Qe(e,r){{var t=function(){he||(he=!0,d(\"%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)\",r))};t.isReactWarning=!0,Object.defineProperty(e,\"ref\",{get:t,configurable:!0})}}var er=function(e,r,t,n,s,u,o){var a={$$typeof:i,type:e,key:r,ref:t,props:o,_owner:u};return a._store={},Object.defineProperty(a._store,\"validated\",{configurable:!1,enumerable:!1,writable:!0,value:!1}),Object.defineProperty(a,\"_self\",{configurable:!1,enumerable:!1,writable:!1,value:n}),Object.defineProperty(a,\"_source\",{configurable:!1,enumerable:!1,writable:!1,value:s}),Object.freeze&&(Object.freeze(a.props),Object.freeze(a)),a};function rr(e,r,t,n,s){{var u,o={},a=null,p=null;t!==void 0&&(de(t),a=\"\"+t),He(r)&&(de(r.key),a=\"\"+r.key),Ke(r)&&(p=r.ref,Je(r,s));for(u in r)D.call(r,u)&&!Xe.hasOwnProperty(u)&&(o[u]=r[u]);if(e&&e.defaultProps){var c=e.defaultProps;for(u in c)o[u]===void 0&&(o[u]=c[u])}if(a||p){var f=typeof e==\"function\"?e.displayName||e.name||\"Unknown\":e;a&&Ze(o,f),p&&Qe(o,f)}return er(e,a,p,s,n,T.current,o)}}var L=w.ReactCurrentOwner,me=w.ReactDebugCurrentFrame;function k(e){if(e){var r=e._owner,t=j(e.type,e._source,r?r.type:null);me.setExtraStackFrame(t)}else me.setExtraStackFrame(null)}var G;G=!1;function B(e){return typeof e==\"object\"&&e!==null&&e.$$typeof===i}function ve(){{if(L.current){var e=b(L.current.type);if(e)return`\n\nCheck the render method of \\``+e+\"`.\"}return\"\"}}function tr(e){{if(e!==void 0){var r=e.fileName.replace(/^.*[\\\\\\/]/,\"\"),t=e.lineNumber;return`\n\nCheck your code at `+r+\":\"+t+\".\"}return\"\"}}var be={};function nr(e){{var r=ve();if(!r){var t=typeof e==\"string\"?e:e.displayName||e.name;t&&(r=`\n\nCheck the top-level render call using <`+t+\">.\")}return r}}function ge(e,r){{if(!e._store||e._store.validated||e.key!=null)return;e._store.validated=!0;var t=nr(r);if(be[t])return;be[t]=!0;var n=\"\";e&&e._owner&&e._owner!==L.current&&(n=\" It was passed a child from \"+b(e._owner.type)+\".\"),k(e),d('Each child in a list should have a unique \"key\" prop.%s%s See https://reactjs.org/link/warning-keys for more information.',t,n),k(null)}}function ye(e,r){{if(typeof e!=\"object\")return;if(M(e))for(var t=0;t<e.length;t++){var n=e[t];B(n)&&ge(n,r)}else if(B(e))e._store&&(e._store.validated=!0);else if(e){var s=Ce(e);if(typeof s==\"function\"&&s!==e.entries)for(var u=s.call(e),o;!(o=u.next()).done;)B(o.value)&&ge(o.value,r)}}}function ar(e){{var r=e.type;if(r==null||typeof r==\"string\")return;var t;if(typeof r==\"function\")t=r.propTypes;else if(typeof r==\"object\"&&(r.$$typeof===P||r.$$typeof===O))t=r.propTypes;else return;if(t){var n=b(r);Ge(t,e.props,\"prop\",n,e)}else if(r.PropTypes!==void 0&&!G){G=!0;var s=b(r);d(\"Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?\",s||\"Unknown\")}typeof r.getDefaultProps==\"function\"&&!r.getDefaultProps.isReactClassApproved&&d(\"getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.\")}}function or(e){{for(var r=Object.keys(e.props),t=0;t<r.length;t++){var n=r[t];if(n!==\"children\"&&n!==\"key\"){k(e),d(\"Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.\",n),k(null);break}}e.ref!==null&&(k(e),d(\"Invalid attribute `ref` supplied to `React.Fragment`.\"),k(null))}}function ir(e,r,t,n,s,u){{var o=Ue(e);if(!o){var a=\"\";(e===void 0||typeof e==\"object\"&&e!==null&&Object.keys(e).length===0)&&(a+=\" You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.\");var p=tr(s);p?a+=p:a+=ve();var c;e===null?c=\"null\":M(e)?c=\"array\":e!==void 0&&e.$$typeof===i?(c=\"<\"+(b(e.type)||\"Unknown\")+\" />\",a=\" Did you accidentally export a JSX literal instead of a component?\"):c=typeof e,d(\"React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s\",c,a)}var f=rr(e,r,t,s,u);if(f==null)return f;if(o){var m=r.children;if(m!==void 0)if(n)if(M(m)){for(var x=0;x<m.length;x++)ye(m[x],e);Object.freeze&&Object.freeze(m)}else d(\"React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.\");else ye(m,e)}return e===_?or(f):ar(f),f}}var sr=ir;z.Fragment=_,z.jsxDEV=sr})()});var Te=q((kr,Re)=>{\"use strict\";Re.exports=xe()});var yr={};pr(yr,{default:()=>gr,frontmatter:()=>vr});var h=hr(Te()),vr={title:\"Android 14\\u2019s storage disaster gets patched, but your data might be gone\",description:\"We get sold technical explainers for all these features, but when they were really needed, none of these poorly thought-out, half-baked systems worked\",image:\"../../public/blogs/android-rage-out.jpeg\",publishedAt:\"2023-11-7\",updatedAt:\"2023-11-7\",author:\"Mark Sikaundi\",isPublished:!0,tags:[\"apple\"]};function Pe(l){let i=Object.assign({p:\"p\"},l.components);return(0,h.jsxDEV)(h.Fragment,{children:[(0,h.jsxDEV)(i.p,{children:`It's the start of November, and that means a new Android security patch. Google claims this one is fixing a high-profile Android 14 storage bug that was locking some people out of their devices. The November Security Bulletin contains the usual pile of security fixes, while consumer-oriented Pixel patch notes list a few user-facing changes. The important line is \"Fix for issue occasionally causing devices with multiple users enabled to show out of space or be in a reboot loop.\" A footnote points out that this is for the \"Pixel 6, Pixel 6a, 6 Pro, 7, 7 Pro, 7a, Tablet, Fold, Pixel 8, Pixel 8 Pro.\"`},void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:13,columnNumber:1},this),`\n`,(0,h.jsxDEV)(i.p,{children:`We're on about day 33 of the Android 14 storage bug. For devices with multiple users set up, there is some kind of storage issue that is locking users out of their device. Some are completely unusable, with the phone bootlooping constantly and never reaching the home screen. Others are able to boot up the device but don't have access to lock storage, which causes a huge amount of issues. Some users likened the bug to \"ransomware,\" a type of malware that encrypts your local storage and then demands money for your data. One fix is to completely erase your device with a factory reset, but a lot of users don't want to do that.`},void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:15,columnNumber:1},this),`\n`,(0,h.jsxDEV)(i.p,{children:\"Android 14\\u2019s user-profile data bug seems indistinguishable from ransomware\"},void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:17,columnNumber:1},this),`\n`,(0,h.jsxDEV)(i.p,{children:\"The earliest reports of this started just days after the October 4 launch date. Google usually rolls updates out slowly so it can pull them if issues like this pop up to minimize damage. That didn't happen here, though. Google failed to respond quickly to initial reports and just let the bug roll out to everyone. Some people even report being freshly hit with the bug just four days ago because Google 1) let the update roll out without stopping it and 2) can't patch its software quickly enough. The biggest issue tracker thread on this bug is up to 1,000-plus likes and 850 comments of people locked out of their devices, and it took two separate rounds of news coverage for Google to acknowledge the bug after about 20 days.\"},void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:19,columnNumber:1},this),`\n`,(0,h.jsxDEV)(i.p,{children:`Google promised a rescue patch, which is now arriving, though the company said only \"some\" data would be recoverable and that \"this update may not enable data to be recovered for devices that are repeatedly rebooting.\" That seems to line up with what people are experiencing now that the OTA is out. People who can boot up report fixed devices, while bootloopers aren't getting anywhere. Some users are still in the support thread hoping for some kind of data recovery for bootlooping devices, but it doesn't seem like that is going to happen.`},void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:21,columnNumber:1},this),`\n`,(0,h.jsxDEV)(i.p,{children:`This whole fiasco has been a complete failure of most of the controls and protections Google has in place in Android. The company slowly rolls out updates to stop problems before it hits a wide number of users, but it failed to pull the update when problems arose. Android has dual system partitions so that you always have a backup if the device fails to boot after an update, but that system didn't work here because Google's \"boot failure\" detection isn't accurate enough. The company shipped a quick-fix patch via Google Play System Updates in the Play Store, but because those passively wait around for a reboot to get applied, users still got hit by the bug days after that patch came out. Android is supposed to have a data backup system for apps, but because that doesn't work well and isn't forced on every app, many users have no backups at all.`},void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:23,columnNumber:1},this),`\n`,(0,h.jsxDEV)(i.p,{children:\"We get sold technical explainers for all these features, but when they were really needed, none of these poorly thought-out, half-baked systems worked. This disaster is a complete technical failure of several Android systems, and many changes need to happen.\"},void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:25,columnNumber:1},this)]},void 0,!0,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\",lineNumber:1,columnNumber:1},this)}function br(l={}){let{wrapper:i}=l.components||{};return i?(0,h.jsxDEV)(i,Object.assign({},l,{children:(0,h.jsxDEV)(Pe,l,void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\"},this)}),void 0,!1,{fileName:\"/Users/marksikaundi/Documents/builds/explore-ui/content/_mdx_bundler_entry_point-36926390-2887-4d76-88e3-616c12760a7f.mdx\"},this):Pe(l)}var gr=br;return mr(yr);})();\n/*! Bundled license information:\n\nreact/cjs/react-jsx-dev-runtime.development.js:\n  (**\n   * @license React\n   * react-jsx-dev-runtime.development.js\n   *\n   * Copyright (c) Facebook, Inc. and its affiliates.\n   *\n   * This source code is licensed under the MIT license found in the\n   * LICENSE file in the root directory of this source tree.\n   *)\n*/\n;return Component;"
  },
  "_id": "android-14s-storage-disaster-gets-patched-but-your-data-might-be-gone/index.mdx",
  "_raw": {
    "sourceFilePath": "android-14s-storage-disaster-gets-patched-but-your-data-might-be-gone/index.mdx",
    "sourceFileName": "index.mdx",
    "sourceFileDir": "android-14s-storage-disaster-gets-patched-but-your-data-might-be-gone",
    "contentType": "mdx",
    "flattenedPath": "android-14s-storage-disaster-gets-patched-but-your-data-might-be-gone"
  },
  "type": "Blog",
  "url": "/blogs/android-14s-storage-disaster-gets-patched-but-your-data-might-be-gone",
  "readingTime": {
    "text": "4 min read",
    "minutes": 3.235,
    "time": 194100,
    "words": 647
  },
  "toc": []
}